core/src/main/scala/dagr/core/tasksystem/Task.scala

```diff
@@ -154,6 +154,9 @@ trait Task extends Dependable {
   // A string set by the enclosing workflow to distinguish the task
   private[tasksystem] var contextName : Option[String] = None
 
+  /** The unique id given to this task by the execution system, or None if not being tracked. */
+  private[core] var id : Option[BigInt] = None
```

Comment:
Doh, I take it back.  Because it's `private[core]` and not just `private` this is in fact an issue.